oar/core/advisory.py

```diff
@@ -305,7 +309,35 @@ def get_doc_prodsec_approved_ads(self):
                 f"get request Docs and Prodsec approved advisories failed"
             ) from e
 
+    def all_advisories_grades_healthy(self):
+        """
+        Check advisories overall grade and advisories image builds grades.
+
+        Returns:
+            bool: True if all advisories grades are healthy
 
+        """
+        healthy_advisories = True
+        
+        for ad in self.get_advisories():
+            if ad.impetus == AD_IMPETUS_RPM:
+                logger.info(f"skipping RPM advisory - {ad.errata_id}, it has no container")
+                continue
+        
+            ad_grade = ad.get_overall_grade()
+            unhealthy_builds = ad.get_unhealthy_builds()
+            
+            if  not util.isGradeHealthy(ad_grade) or unhealthy_builds:
```

Comment:
the logic should be, if overall grade is less than B, then check unhealthy build for ths advisory

Reply:
Fixed. Please check.